pinot-core/src/test/java/org/apache/pinot/core/realtime/stream/StreamConfigTest.java

```diff
@@ -279,6 +285,16 @@ public void testStreamConfigValidations() {
     streamConfig = new StreamConfig(tableName, streamConfigMap);
     assertEquals(streamConfig.getFlushThresholdSegmentSizeBytes(),
         StreamConfig.DEFAULT_FLUSH_THRESHOLD_SEGMENT_SIZE_BYTES);
+
+    // If size based threshold is set, then rows must be 0
+    streamConfigMap.put(StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_ROWS, "1000000");
+    streamConfigMap.put(StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_SEGMENT_SIZE, "100M");
+    try {
+      new StreamConfig(tableName, streamConfigMap);
+      fail("Invalid config: flush threshold rows must be 0, when flush threshold size is set.");
+    } catch (Exception e) {
+      // Expected
+    }
```

Comment:
i am not sure this is what the document intended to mean
what i read was:
```
You can pick the appropriate value for segment size and number of hours in the table config, and set the number of rows to zero. Note that you don't have to pick values exactly as given in each of these combinations (they are calculated guesses anyway).
```
it wouldn't throw exception if num rows is set to non-zero. 

my concern is that this will cause issue during cluster upgrade. if any user have both configuration set table will be unable to load in an error state .